pkgs/by-name/re/regripper/package.nix

```diff
@@ -0,0 +1,56 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, perl
+, perlPackages
+, runtimeShell
+}:
+
+stdenv.mkDerivation rec {
+  pname = "regripper";
+  version = "unstable-2023-07-23";
+
+  src = fetchFromGitHub {
+    owner = "keydet89";
+    repo = "RegRipper3.0";
+    rev = "cee174fb6f137b14c426e97d17945ddee0d31051";
+    sha256 = "sha256-vejIRlcVjxQJpxJabJJcljODYr+lLJjYINVtAPObvkQ=";
+  };
+
+  buildInputs = [ perl perlPackages.Parse-Win32Registry ];
```

Comment:
We'll need the module at runtime as well, as otherwise we can't run the `regripper` script:

```sh
[nix-shell:~/.cache/nixpkgs-review/pr-278373-1]$ regripper 
Can't locate Parse/Win32Registry.pm in @INC (you may need to install the Parse::Win32Registry module) (@INC entries checked: /nix/store/6icf13zdmp9wkxc4n299vix9fsmx0jw7-perl-5.38.2/lib/perl5/site_perl/5.38.2/darwin-thread-multi-2level /nix/store/6icf13zdmp9wkxc4n299vix9fsmx0jw7-perl-5.38.2/lib/perl5/site_perl/5.38.2 /nix/store/6icf13zdmp9wkxc4n299vix9fsmx0jw7-perl-5.38.2/lib/perl5/5.38.2/darwin-thread-multi-2level /nix/store/6icf13zdmp9wkxc4n299vix9fsmx0jw7-perl-5.38.2/lib/perl5/5.38.2) at /nix/store/4lzl5lk7b13l5gk43yjkfwjz6nx06a5c-regripper-unstable-2023-07-23/share/regripper/rip.pl line 34.
BEGIN failed--compilation aborted at /nix/store/4lzl5lk7b13l5gk43yjkfwjz6nx06a5c-regripper-unstable-2023-07-23/share/regripper/rip.pl line 34.
```

```suggestion
  propagatedBuildInputs = [ perl perlPackages.Parse-Win32Registry ];
```

Might also consider to make a `perlEnv` wrapper instead like in https://github.com/NixOS/nixpkgs/blob/ef73683e6c3807bae7d97128b2b27373239fd8a4/pkgs/tools/security/regpg/default.nix#L10 so `regripper` can run a self-contained Perl environment.

Reply:
@zakame if I use `perlEnv` approach, it will fail if someone wants to test it, because `Parse-Win32Registry` is still not inside nixpkgs repository.

Reply:
Right - we normally factor out the dependent package into its own PR, e.g `perlPackages.ParseWin32Registry: init at 1.1`.

Also I noticed this package name is wrong, there should be no `-`.

Reply:
Oh ok. I was thinking that missing dependencies should be inside the same PR with a commit named `perlPackages.ParseWin32Registry: init at 1.1`.

Reply:
@zakame my doubt is that: if I do separated PRs between the main program and its dependency, the `Check pkgs/by-name / check (pull_request_target)` will fail because in the current forked repository at that specific branch the dependency cannot be found. Instead, if I insert program and dependency in the same PR, this check should not fail because these two packages live in the same forked repository in the same branch.

Reply:
I don't think a separate PR is required here, since `perlPackages.ParseWin32Registry` is a dependency for regripper.